### PR TITLE
fix: Modify PEG rule to support escape chracter

### PR DESCRIPTION
**Pull request by tkxkd0159**
## Description

Closes: #773

### CODE REVIEW

**Review by jaeseung-bae — APPROVED** [review]

**Review by zemyblue — APPROVED** [review]

File: libs/pubsub/query/query.peg
Comment (ulbqb) [inline libs/pubsub/query/query.peg:18]:
```
value <- < '\'' (!['] .)* '\''>
```
Why not this?

Comment (tkxkd0159) [reply libs/pubsub/query/query.peg:18]: We need to support Go escape character such as `\"` because of JSON unmarshaled string.

Comment (Mdaiki0730) [reply libs/pubsub/query/query.peg:18]: In this case, how about limiting the escape character to `"`?
```suggestion
value <- < '\'' (('\\' '\"') / (!['] .))* '\''>
```

Comment (ulbqb) [reply libs/pubsub/query/query.peg:18]: `\"` is one byte data, but you mean data to unmarshal is two bytes data?

Comment (tkxkd0159) [reply libs/pubsub/query/query.peg:18]: @ulbqb @Mdaiki0730 sorry for typo in my previous answer. We need to accept any character if it's escaped.

**Review by Mdaiki0730 — APPROVED** [review]

File: libs/pubsub/query/query.peg
Comment (tkxkd0159) [reply libs/pubsub/query/query.peg:18]: I will merge this PR after @ulbqb approve

Comment (ulbqb) [reply libs/pubsub/query/query.peg:18]: Actual raw data to unmarshal doesn't contain `\` as escape symbol even if it contains `"` as text. 
Previously, `\` as text is not allowed so your PR contains:

- allow `\` as text
- allow `"` as text

Is this correct?

Comment (tkxkd0159) [reply libs/pubsub/query/query.peg:18]: > Actual raw data to unmarshal doesn't contain \ as escape symbol even if it contains " as text.
Yes, that's right
> Your PR contains:
Yes, that's correct. So if you want to add `'` in event string. you need to put `\\'` instead of `'`.

**Review by ulbqb — APPROVED** [review]

**Pull request merged**
